builder/dockerignore/dockerignore.go

```diff
@@ -20,6 +20,10 @@ func ReadAll(reader io.ReadCloser) ([]string, error) {
 	var excludes []string
 
 	for scanner.Scan() {
+		// Lines starting with # (comments) are ignored before processing
+		if strings.HasPrefix(scanner.Text(), "#") {
```

Comment:
its a minor performance (stylistic) thing, but can you change this so that its:
```
pattern := scanner.Text()
if strings.HasPrefix(pattern, "#") { ... }
pattern = strings.TrimSpace(pattern)
```
So that we're not calling scanner.Text() more than once?